serverless/resource_deployment.go

```diff
@@ -0,0 +1,273 @@
+package serverless
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"golang.org/x/mod/sumdb/dirhash"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/cloudformation"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/customdiff"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+type serverlessConfig struct {
+	Service string
+}
+
+func getServerlessConfig(configDir string, serverlessBinDir string) ([]byte, error) {
+	cmd := exec.Command(getServerlessBin(serverlessBinDir), "print", "--format", "json")
+	cmd.Dir = configDir
+
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return []byte{}, fmt.Errorf("%v\n%w", string(output), err)
+	}
+
+	return output, nil
+}
+
+func getServiceName(configJson []byte) (string, error) {
+	config := serverlessConfig{}
+	err := json.Unmarshal([]byte(configJson), &config)
+
+	return config.Service, err
+}
+
+func hashServerlessDir(configDir string, packagePath string, configJson []byte) (string, error) {
+	absolutePackagePath := filepath.Join(configDir, packagePath)
+	zipPath := filepath.Join(absolutePackagePath, "sls-provider.zip")
```

Comment:
This looks hard-coded. Isn't this supposed to be based on `getServiceName`?

Reply:
Also, what happens if `package.individually = true` is set and we get lots of zips?

Reply:
Whew glad you caught that hardcoding 😅 for the `package.individually`, I'd like to ticket it our for a follow-up PR. I imagine we can hash all of the zip files in `package_dir` to solve it!

Reply:
Ticket for `package.individually`: https://github.com/FormidableLabs/terraform-provider-serverless/issues/3